### PR TITLE
feature: Add Customer Group resource

### PR DESCRIPTION
**Pull request by IrmantasM**
https://github.com/labd/terraform-provider-commercetools/issues/103

My first submission + never used golang before, so don't be too harsh 😄

**This change includes:**
- [x] Added Customer Group resource
- [x] Added Customer Group resource acceptance tests
- [x] Added missing parameters to `local/testenv.sh.example`

Acceptance Tests pass and I have also tested manually on my personal CommerceTools project, seems to be working okay but would be good to test more extensively, especially if someone knows some edge cases related to Customer Groups.

There are failing tests for Discount Code resource, not sure how to fix it, I didn't touch that part: https://github.com/labd/terraform-provider-commercetools/issues/79

**Added new groups:**
image

**Changed attributes in groups:**
image

**Removed group:**
image

### CODE REVIEW

File: commercetools/resource_customer_group.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		log.Fatal("No customer group")
+	}
+
+	d.SetId(customerGroup.ID)
+	d.Set("version", customerGroup.Version)
+
+	return resourceCustomerGrouptRead(d, m)
+}
+
+func resourceCustomerGrouptRead(d *schema.ResourceData, m interface{}) error {
```
Comment (davidweterings) [inline commercetools/resource_customer_group.go:72]: I totally missed this PR 😓 , can you remove the 't' after Group?

File: commercetools/resource_customer_group.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+	if d.HasChange("name") {
+		newName := d.Get("name").(string)
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CustomerGroupChangeNameAction{Name: newName})
+	}
+
+	if d.HasChange("key") {
+		newName := d.Get("key").(string)
```
Comment (davidweterings) [inline commercetools/resource_customer_group.go:125]: This should really be called `newKey` to avoid confusion with the code above

File: commercetools/resource_customer_group.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	version := d.Get("version").(int)
+	_, err := client.CustomerGroupDeleteWithID(d.Id(), version)
+	if err != nil {
+		log.Printf("[ERROR] Error during deleting customer group resource %s", err)
+		return nil
+	}
+	return nil
+}
+
+func resourceCustomerGroupGetGroups(d *schema.ResourceData) []string {
```
Comment (davidweterings) [inline commercetools/resource_customer_group.go:157]: Can you remove the last two functions, it seems they are not being used unless I'm missing something 😅 ?

**Comment by davidweterings** [comment]
@IrmantasM looks great, especially if this is your first go code 🎉 ! I have some small remarks, can you solve these?

**Comment by IrmantasM** [comment]
Thanks for review @davidweterings ! Will update sometime this week.

**Comment by mvantellingen** [comment]
@IrmantasM do you still have interest in picking this up?

**Comment by qdegraaf** [comment]
Continuing this work in https://github.com/labd/terraform-provider-commercetools/pull/141

**Comment by qdegraaf** [comment]
This was done in https://github.com/labd/terraform-provider-commercetools/pull/141 . Closing
